vm/regexp.go

```diff
@@ -0,0 +1,139 @@
+package vm
+
+import (
+	"strconv"
+
+	"github.com/dlclark/regexp2"
+	"github.com/goby-lang/goby/vm/classes"
+	"github.com/goby-lang/goby/vm/errors"
+)
+
+// RegexpObject represents regexp instances, which of the type is actually string.
+// Regexp object holds regexp strings.
+// Powered by: github.com/dlclark/regexp2
+// The regexp2 package had been ported from .NET Framework's regexp library, which is PCRE-compatible
+// and is almost all equivalent to Ruby's Onigmo regexp library.
+//
+// ```ruby
+// a = Regexp.new("orl")
+// a.match?("Hello World")   #=> true
+// a.match?("Hello Regexp")  #=> false
+//
+// b = Regexp.new("😏")
+// b.match?("🤡 😏 😐")   #=> true
+// b.match?("😝 😍 😊")   #=> false
+//
+// c = Regexp.new("居(ら(?=れ)|さ(?=せ)|る|ろ|れ(?=[ばる])|よ|(?=な[いかくけそ]|ま[しすせ]|そう|た|て))")
+// c.match?("居られればいいのに")  #=> true
+// c.match?("居ずまいを正す")      #=> false
+// ```
+//
+// **Note:**
+//
+// - Currently, manipulations are based upon Golang's Unicode manipulations.
+// - Currently, UTF-8 encoding is assumed based upon Golang's string manipulation, but the encoding is not actually specified(TBD).
+// - `Regexp.new` is exceptionally supported.
+//
+// **To Goby maintainers**: avoid using Go's standard regexp package (slow and not rich). Consider the faster `Trim` or `Split` etc in Go's "strings" package first, or just use the dlclark/regexp2 instead.
+// ToDo: Regexp literals with '/.../' and match operator `=~`
+type RegexpObject struct {
+	*baseObj
+	Regexp *regexp2.Regexp
+}
+
+// Class methods --------------------------------------------------------
+func builtInRegexpClassMethods() []*BuiltinMethodObject {
+	return []*BuiltinMethodObject{
+		{
+			Name: "new",
+			Fn: func(receiver Object) builtinMethodBody {
+				return func(t *thread, args []Object, blockFrame *callFrame) Object {
```

Comment:
Does this need a check for the number of arguments? (I'm not a 100% sure if the parser does this check or not)

Reply:
I think this method should check argument numbers, and check it here.
@hachi8833 Can you make sure only one argument should be passed into this method?

Reply:
@hachi8833 You should check the argument's number and its type in here, before you initialize and regexp object

Reply:
Fixed